src/typescript/azure-openapi-validator/rules/Rpaas_CreateOperationAsyncResponseValidation.ts

```diff
@@ -0,0 +1,60 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { MergeStates, OpenApiTypes, rules } from "../rule"
+export const Rpaas_CreateOperationAsyncResponseValidation: string = "Rpaas_CreateOperationAsyncResponseValidation"
+
+rules.push({
+  id: "R4023",
+  name: Rpaas_CreateOperationAsyncResponseValidation,
+  severity: "error",
+  category: "RPaaSViolation",
+  mergeState: MergeStates.individual,
+  openapiType: OpenApiTypes.rpass,
+  appliesTo_JsonQuery: "$.paths.*.put",
+  *run(doc, node, path) {
+    if (node.responses["202"]) {
+      yield {
+        message: `[RPaaS] Only 201 is the supported response code for PUT async response.`,
+        location: path
+      }
+    }
+
+    const isAsyncOperation = node.responses["201"]
+      || (node["x-ms-long-running-operation"] && node["x-ms-long-running-operation"] == true)
```

Comment:
Is this condition (201) enough to indicate a async operation ?  As I scan all the swaggers in the repo, produced almost 2000 errors about this rule

Reply:
Yes. 201 is async and status is tracked using a property (Properties.ProvisioningState) that must reach terminal state. 
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creatingupdating-using-put

Reply:
But RPs mostly missed annotating it with x-ms-long-running-operation and -options. 
This rule should run for RPaaS RPs only.